server/routes/badge.routes.js

```diff
@@ -129,4 +133,33 @@ router.get('/all', authenticate, isAdmin, BadgeController.getAllBadges);
  */
 router.get('/:id', authenticate, hasValidObjectId, isAdmin, BadgeController.getOneBadge);
 
+/**
+ * @swagger
+ * /badge/role-specific:
+ *   post:
+ *     tags:
+ *       - Badge
+ *     description: Returns all badges and role specific badge
+ *     produces:
+ *       - application/json
+ *     requestBody:
+ *      content:
+ *        application/json:
+ *          schema:
+ *            $ref: '#/components/schemas/Badge'
+ *      description: Returns all badges and role specific badge
+ *     responses:
+ *      '200':
+ *        description: Badges returned
+ *      '500':
+ *       description: Internal server error
+ */
+router.post(
```

Comment:
Following the best convention, I think this is supposed to be a `get` not a `post` method. We can add the needed role as a query parameter.
```suggestion
router.get(
```